metriq_gym/benchmarks/bseq.py

```diff
@@ -0,0 +1,239 @@
+        The graph of edges that violated the CHSH inequality.
+    """
+    # A subgraph is constructed containing only the edges (qubit pairs) that successfully violate the CHSH inequality.
+    # The size of the largest connected component in this subgraph provides a measure of the device's performance.
+    good_edges = []
+    for job_idx, result in enumerate(result_data):
+        if result.measurement_counts is None:
+            continue
+        num_meas_pairs = len(
+            {key for key, val in coloring.edge_color_map.items() if val == job_idx}
+        )
+        exp_vals: np.ndarray = np.zeros(num_meas_pairs, dtype=float)
+        for idx in range(4):
+            counts = result.measurement_counts[idx]
+            # Expectation values for the CHSH correlation terms are calculated based on measurement outcomes.
+            for pair in range(num_meas_pairs):
+                sub_counts = marginal_counts(counts, [2 * pair, 2 * pair + 1])
+                exp_val = sampled_expectation_value(sub_counts, "ZZ")
+                exp_vals[pair] += exp_val if idx != 2 else -exp_val
+        for idx, edge_idx in enumerate(
+            key for key, val in coloring.edge_color_map.items() if val == job_idx
+        ):
+            edge = (coloring.edge_index_map[edge_idx][0], coloring.edge_index_map[edge_idx][1])
+            # The benchmark checks whether the CHSH inequality is violated (i.e., the sum of correlations exceeds 2,
+            # indicating entanglement).
+            if exp_vals[idx] > 2:
+                good_edges.append(edge)
+    good_graph = rx.PyGraph(multigraph=False)
+    good_graph.add_nodes_from(list(range(coloring.num_nodes)))
+    for edge in good_edges:
+        good_graph.add_edge(*edge, 1)
+    return good_graph
+
+
+def largest_connected_size(good_graph: rx.PyGraph) -> int:
+    """Finds the size of the largest connected component in the CHSH subgraph.
+
+    Args:
+        good_graph: The graph of qubit pairs that violated CHSH inequality.
+
+    Returns:
+        The size of the largest connected component.
+    """
+    cc = rx.connected_components(good_graph)
+    largest_cc = cc[np.argmax([len(g) for g in cc])]
+    return len(largest_cc)
+
+
+class BSEQ(Benchmark):
+    """Benchmark class for BSEQ (Bell state effective qubits) experiments."""
+
+    def dispatch_handler(self, device: QuantumDevice) -> BSEQData:
+        """Runs the benchmark and returns job metadata."""
+        shots = self.params.shots
+
+        topology_graph = None
+        coloring = None
+        trans_exp_sets: list[list[QuantumCircuit]]
+
+        if isinstance(device, QiskitBackend):
+            coloring = ibm_device_coloring(device)
+            exp_sets = generate_chsh_circuit_sets(coloring)
+            pm = generate_preset_pass_manager(1, device._backend)
+            trans_exp_sets = [pm.run(circ_set) for circ_set in exp_sets]
+        else:
+            raise ValueError(f"Unsupported device type: {type(device)}")
```

Comment:
I know we are soon going to implement this for other platforms. However, we should start thinking about a mechanism to see the 'unsupported scenario' as a test skip rather than a test fail (as suggested by @nonhermitian). I will create a ticket about this.